### DISCUSSION TITLE
Auto-archive items in your projects

### DISCUSSION THREAD

**Discussion by 2percentsilk** [community/community#36466] [Projects and Issues]
We've just released auto-archived for your projects to help keep you boards and table tidy and focused on what's most important now. No longer will you need to set aside time to select all closed items and archive them.

Check out [how to set up your workflow](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/archiving-items-automatically) and give us feedback as you try it out!

https://user-images.githubusercontent.com/3174849/196257267-d919b138-ddb1-464c-b295-62341e1b2b0a.mp4

**Comment by tay1orjones** [comment]
I'm so glad to see this workflow become available! 🎉  

While `last-updated` is nice, I'd like to instead automatically archive items based on what iteration/sprint they were included in.

### Use case

Previously, before this workflow was available, we set up a [view](https://github.com/orgs/carbon-design-system/projects/39/views/49) for items that are ready to be archived. The filter it uses only gathers closed or merged items that are 3 sprints or older (or have no sprint):

```
is:closed,merged -sprint:<@current,@previous,@next,@previous-1,@previous-2 
```

Since insights charts do not include archived items, auto-archiving based on this filter is crucial to retain usable data in our charts for planning purposes. For instance, we track the total count of issues from our past three sprints in attempt to forecast how much we should plan for the next sprint. We've clearly over-planned our current Sprint 46! 😅 

![image](https://user-images.githubusercontent.com/3360588/196513456-3fc53105-18f7-4e17-be5d-1e869a751678.png)

**Reply by tay1orjones** [reply]
Also sort of related here https://github.com/community/community/discussions/36584, it would be nice if the default workflow could be duplicated so we could have two instances of this workflow:

```
is:closed,merged -sprint:<@current,@previous,@next,@previous-1,@previous-2 
```

and

```
is:closed,merged reason:not-planned
```

So that `reason:not-planned` issues would be automatically archived pretty much immediately on close and not have to wait 56 days after they're closed.

**Reply by 2percentsilk** [reply]
Duplicating workflows is definitely something I'm hoping to work on bringing to customers soon 👀 

For the iteration one, we actually considered allowing that as a modifier...question for you: How do you think we should deal with timezones in this case? Should we check for the iteration the day after it ends and run the workflow? How would you expect this to work on the cusps of iterations where the iteration may be over for teammates in Australia but it's the last day of the iteration for folks in California?

**Reply by archon810** [reply]
A side question - is `is:closed,merged` necessary instead of just `is:closed`? Would there ever be a `merged` that isn't `closed`?

**Reply by 2percentsilk** [reply]
@archon810
`closed` will match pull requests that are closed without merging (as well as any closed issues), while `merged` explicitly refers to pull requests. `is:closed, merged` will give you the union of merged pull requests AND closed issues AND closed without merged pull requests. So it really depends on what you're trying to accomplish! In some cases, it doesn't matter (aka when you're already scoped to PRs) but it does matter if your workflow looks at issues AND PRs.

**Reply by alexanderniebuhr** [reply]
@2percentsilk we do have the same requirement, regarding iterations. As far as timezone concern, I think either two possibilities could be great:

- let users define a management timezone in the setting of the projects, which this workflow is based on
- run the workflow when the last timezone on the world has finished the iteration, there has to be a way to tell which is the latest timezone

_maybe at least allow the modifier with a information about the issue until a perfect solution is found, i rather have the modifier with timezone issue, than not having it at all_

**Reply by tay1orjones** [reply]
@2percentsilk I'm not too concerned about the workflow running exactly when the iteration ends. As @alexanderniebuhr mentioned, an option would be nice but honestly in my mind the iteration isn't over until it's over for _everyone_. 

Running it the day after the iteration ends would be fine, or when the last timezone hits and the day is officially over. [IDLW, `UTC−12:00`](https://en.wikipedia.org/wiki/UTC%E2%88%9212:00) is the "last" time zone and is ~8hrs behind NYC.

**Reply by iPwnPancakes** [reply]
Apologies for reviving this older feature request, but I found myself wanting this exact same thing. 

Regarding the timezone issue, the only idea I can come up with is adding a particular timezone field to the project in the project settings and giving users the option to view the project as a different timezone than the one configured

**Comment by lfrancke** [comment]
Thanks for this feature and congratulations on shipping! :tada: 

I have one minor issue.
![image](https://user-images.githubusercontent.com/122850/196896247-3b2860e9-d8e5-477f-9b70-45eb04156297.png)

This filter works fine in my board (status is a custom field) but when I use the same filter in the auto archive box I get this:

![image](https://user-images.githubusercontent.com/122850/196896668-23cd23ad-2d45-483e-9e79-25c79db9597c.png)

As you can see: It still shows the correct amount (3 items fit this filter) but it complains that the filter is invalid.
I _assume_ it's going to work but the warning is confusing.

**Reply by ghost** [reply]
I would also like to see `status` supported in the auto-archiving feature.

**Reply by sercanuste** [reply]
+1

**Reply by lfrancke** [reply]
I can now report that this is broken. It archived all our things across all "status" fields - so it executed part of the filter and ignored the thing it deemed invalid. This is bad @2percentsilk

**Reply by 2percentsilk** [reply]
Thanks for the report and sorry this happened :( 

The filter shouldn't run at all if a piece is invalid so I'll get that filed and fixed.

While we don't have an undo button, if you go to the archive page, you'll be able to select your items and restore them to the project board with their custom field values they had previously.

**Reply by lfrancke** [reply]
Thank you!

**Comment by lfrancke** [comment]
Apart from the issue I mentioned in my other thread I have one other point of feedback: 

It's a tiny bit confusing to have the "When" filter at the top which only lists _issue_ and _pull request_ but not _drafts_, especially when the whole thing can be done with a filter query anyway:

![image](https://user-images.githubusercontent.com/122850/196898025-053da691-52c7-43c6-a352-700c41e189d4.png)

I believe selecting `issue` at the top is equivalent to this filter `is:draft,issue`. 
Wouldn't it make more sense to separate them or to leave out the _When_ bit entirely?

**Comment by hCraker** [comment]
Hello,
My team is using the new project board view. It has been working great for us, but we've had some issues with issues being auto archived despite that workflow being turned off in our project settings. Has anyone else encountered this since the auto-archive  feature was added?

Thanks,
Heather

**Comment by joshuacwnewton** [comment]
Hi there! I'd like to use the auto-archive workflow with the `is:closed` status, but only for issues that are closed as "Not planned". In other words, I'd like to preserve the issues that are closed as "Completed", because it's useful to display them in a `Done` status column for release planning.

![image](https://user-images.githubusercontent.com/16181459/217687195-b5299566-a500-4695-8a9e-ae0f0a7327b6.png)

I can't seem to find a way to distinguish between the two types of `is:closed`, though. Is there a way?

**Reply by 2percentsilk** [reply]
`is:closed, reason:"not planned"` should do that for you!

**Reply by joshuacwnewton** [reply]
Thank you so much! I just tested this out and it works perfectly. 

Apologies for the noise on a (somewhat) unrelated discussion thread!

**Comment by rileybroughten** [comment]
Adding feedback from another [discussion](https://github.com/orgs/community/discussions/49722) from @doronguttman:

The auto-archive feature on projects is great!
But it would be better if it could only archive items of a certain status (e.g. status:Done) - currently the status filter is not supported.

**Reply by blaxphoenix** [reply]
It would be good indeed. In my case, this would help archive only "Done" tickets, not all closed tickets. Is there any info on when this would be available? Thanks! 😄

**Reply by marcantoine** [reply]
This feature would indeed be useful since we cannot prevent PR to close issues: https://github.com/orgs/community/discussions/66741 

Once PR are merged, this workflow archives the issues after a few days, even though the status is still in QA and not even in production yet.

**Comment by goslackware** [comment]
Is this feature available in enterprise server (self hosted)?  If so, what version was or will it be added to?

**Reply by rileybroughten** [reply]
@goslackware - This (along with auto-add workflow) will be available in GHES 3.9 👍

**Comment by healthbjk** [comment]
The documentation suggests that only these filters are supported:
"You can only use the is, reason, and last-updated filters."

However, it sorta acts like it recognizes `updated`:
<img width="899" alt="Screen Shot 2023-05-02 at 12 39 25 AM" src="https://user-images.githubusercontent.com/68571760/235580585-cc5653e8-8b1a-4f92-bcf4-34032fdbc73e.png">

Are there undocumented filters?

**Reply by lfrancke** [reply]
This looks to be the same as https://github.com/orgs/community/discussions/36466#discussioncomment-3922308

**Reply by healthbjk** [reply]
It appears different to me. The filter appears as valid:

<img width="1330" alt="image" src="https://user-images.githubusercontent.com/68571760/235766627-588dab08-398d-408e-adf4-7182332c27aa.png">

**Reply by reneexeener** [reply]
`last-updated` is only used inside Projects so we added support for `updated` to make it more consistent with the rest of GH search. We encourage users to use `updated`. I will let the team know that the doc needs to be updated. Thanks for reporting this!

**Comment by parberge** [comment]
We want to auto archive pull requests that are closed and not merged.
Tried with this filter `is:draft,pr is:closed is:unmerged`, which works if you use it in the pull request page.
However using it in the auto-archive workflow we get the following message:
<img width="639" alt="image" src="https://github.com/community/community/assets/5441497/89ede42b-63fb-4975-a84b-1197499522c5">

Even though it warned us we could still use the query, but the result was ofc not what we wanted (it archived pull requests that was closed and merged).

Is there a way to auto archive closed unmerged PRs?

**Reply by parberge** [reply]
Oh seems like this is the correct query to use: `is:draft,pr is:closed -is:merged`.

**Comment by dinaolmelak** [comment]
Hello 👋 ,
I would like to start by expressing my gratitude for this awesome feature. I just wanted to mention the recent bother I discovered. I realized that it takes its sweet time for archiving. 
<img width="1406" alt="Screenshot 2023-08-02 at 10 22 26 PM" src="https://github.com/community/community/assets/32272045/9b7998f3-eebd-4cd0-ba17-d54ffc0ae806">

You can see the filter has been triggered but it is not archiving it. Am I missing something?
